src/ssl.c

```diff
@@ -4953,6 +4957,14 @@ int wolfSSL_Init(void)
 #endif
 
 #ifdef OPENSSL_EXTRA
+        if (wc_InitMutex(&rand_methods_mutex) != 0) {
```

Comment:
This code section is duplicated about six times. Can you please add a static local function for this to reduce code size? Also if you call this new static local and directly return on an error response then you can eliminate all the `if (initRandMethMutex != 0) {` logic on each of these.

Reply:
I will move this logic into a function but to be clear, I thought you didn't want any early returns in these functions.  Are you ok with me returning early in this case?

Reply:
I am very okay with a return in the middle of the function. My suggestion was to avoid having multiple unlock paths.